src/toil/batchSystems/singleMachine.py

```diff
@@ -151,28 +157,52 @@ def worker(self, inputQueue):
                                   jobCores)
                         with self.coreFractions.acquisitionOf(coreFractions):
                             with self.disk.acquisitionOf(jobDisk):
-                                startTime = time.time() #Time job is started
-                                with self.popenLock:
-                                    popen = subprocess.Popen(jobCommand,
+                                startTime = time.time()  # Time job is started
```

Comment:
Could this be split into multiple functions? Function is way too long and nested